### PR TITLE
Support: support token survives refresh

### PR DESCRIPTION
**Pull request by rodrigoi**
This PR prevents support token to be cleared on refresh. More information, testing instructions and security concerns here: p7jreA-1sg-p2

### CODE REVIEW

**Comment by kwight** [comment]
Gave this a run, looks great; logging out from the supported user works well too.

File: client/lib/support/support-user/localstorage-bypass.js [hunk 1]
```diff
@@ -84,11 +84,11 @@ export default function( allowedKeys ) {
 		const _getItem = window.Storage.prototype.getItem.bind( window.localStorage );
 		const _removeItem = window.Storage.prototype.removeItem.bind( window.localStorage );
 
-		Object.defineProperty( Storage.prototype, 'length', {
```
Comment (gwwar) [inline client/lib/support/support-user/localstorage-bypass.js]: Note that Storage is used as an interface for session and local storage. 
https://developer.mozilla.org/en-US/docs/Web/API/Storage

No other place explicitly uses session storage except for social login. (Everything else should be using localForage and whatever compatible driver the user's browser has)

File: client/lib/user/support-user-interop.js [hunk 2]
```diff
@@ -78,7 +77,7 @@ const _isSupportUserSession = ( () => {
 		return false;
 	}
 
-	const supportUser = store.get( STORAGE_KEY );
+	const supportUser = JSON.parse( sessionStorage.getItem( STORAGE_KEY ) );
```
Comment (gwwar) [inline client/lib/user/support-user-interop.js]: JSON.parse might throw. Let's add a try catch, where we return false in the catch.

File: client/lib/user/support-user-interop.js [hunk 3]
```diff
@@ -132,20 +144,22 @@ export const boot = () => {
 		return;
 	}
 
-	localforage.bypass();
-
-	const { user, token } = store.get( STORAGE_KEY );
+	const { user, token } = JSON.parse( window.sessionStorage.getItem( STORAGE_KEY ) );
```
Comment (gwwar) [inline client/lib/user/support-user-interop.js]: Let's also add a try catch here, with an early return in the catch.

File: client/lib/user/support-user-interop.js (hunk 2 above)
Comment (rodrigoi) [reply client/lib/user/support-user-interop.js]: updated on 04e33a972fe2a8530051d8d2c12b13850acca81b.

File: client/lib/user/support-user-interop.js (hunk 3 above)
Comment (rodrigoi) [reply client/lib/user/support-user-interop.js]: updated on 04e33a972fe2a8530051d8d2c12b13850acca81b.

**Comment by gwwar** [comment]
Unless @vortfu or @xknown had any other requests, let's try landing this on Monday.

**Pull request merged**